homeassistant/components/evohome/__init__.py

```diff
@@ -214,7 +137,7 @@ async def load_auth_tokens(store) -> tuple[dict, dict | None]:
     try:
         await client_v2.login()
     except (aiohttp.ClientError, evohomeasync2.AuthenticationError) as err:
```

Comment:
Since I see that you're the codeowner of `evohome-async`, we should strive to have client libraries abstract exceptions from their dependencies – could you perhaps generalize a `evohomeasync2.EvoHomeError` and catch only that?

Reply:
Yes - I am the codeowner for the async version of the venerable **requests** version (**evohomeclient**, which I am not CO for).

To make porting to async easy, I primary goal was to minimize any differences between the two, and am equally reluctant to be backporting...

Reply:
I have had a think about this...

> we should strive to have client libraries abstract exceptions from their dependencies

Sure? I feel strongly that it is inappropriate to wrap `aiohttp.ClientError` with a derivation of `evohomeasync2.EvohomeError`. `ClientError` is familiar & well-understood and calling it something else will only serve to create confusion for little benefit.

In addition, I would be having to re-raise `evohomeasync2.ClientError` as either:
 - `UpdateFailed` or
 - `aiohttp.ClientError` (better)
... to provide the correct context for `DataUpdateCoordinator._async_refresh()`

I think inheriting `evohomeasync2.AuthenticationError` from `evohomeasync2.EvoHomeError` is OK, but then I'm only interested in `evohomeasync2.AuthenticationError`, which I'll be re-raising (`raise ... from AuthenticationError`) as either:
 - `ConfigEntryAuthFailed` during integration setup (i.e. bad user credentials), or
 - `UpdateFailed` during refresh (i.e. authorization failed)

Reply:
> `ClientError` is familiar & well-understood and calling it something else will only serve to create confusion for little benefit.

That isn't universally true. I would contend the reverse: littering the HASS codebase with `aiohttp`/`httpx`/etc. errors reinforces the (incorrect) belief that HASS should be aware of HTTP op internals. The pattern I'm describing is used in most integrations, and I've yet to see it create confusion.

Reply:
I am sorry, I have decided that refactoring **evohome-async** is not something I'm willing to do:
 - it is used by platforms other than HA
 - it is intended to be a faithful async port of **evohome-client** (i.e. same APIs, Exceptions, albeit for aiohttp rather than requests) - I would then need to backport
 - I don't feel it really adds sufficient value to this integration to do so
 
Can you see any way to proceed without refactoring **evohome-async**?
 
Alternatively, the whole idea here was to get config flow into evohome, this was just the first step. Can I do that without using data coordinator?

Reply:
Have you read my notes below about the contextualisation - almost all of it is at the integration layer anyway.

Reply:
Happy to mark this as needing a second opinion. I appreciate your different opinion, but what I'm requesting is the general standard required of HASS integrations. Perhaps another reviewer can offer other thoughts.

Reply:
Thanks for your input.

Reply:
Ideally we have named exceptions, but since the coordinator is already [aware](https://github.com/home-assistant/core/blob/83a6012f8061ef8c5c8d5f60cb0ded7282d0dd1e/homeassistant/helpers/update_coordinator.py#L214) of `(aiohttp.ClientError, requests.exceptions.RequestException)`, its reasonable to be catching them.

We shouldn't do additional logging while updating.  Instead of the `_contextualise...` wrap the error message in `UpdateFailed` and raise it.   The coordinator will take care of if it should be logged or not based on silver quality scale (https://developers.home-assistant.io/docs/integration_quality_scale_index?_highlight=logs#silver-)

- Handles internet unavailable. Log a warning once when unavailable, log once when reconnected.
- Handles device/service unavailable. Log a warning once when unavailable, log once when reconnected.

You can probably save yourself a bit of refactoring by subclassing the `DataUpdateCoordinator`

If you have it all hooked up and converted to a config entry you should be able to do `await coordinator.async_config_entry_first_refresh()` in your `async_setup_entry` and the error will bubble up and appear in the UI if setup fails.  You'll have to convert to a config entry to make that work though.